### ISSUE TITLE
`aws_rds_cluster` `engine` argument not supported

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#2060]
_This issue was originally opened by @ajenkins-cargometrics as hashicorp/terraform#16453. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

The documentation for the `aws_rds_cluster` resource lists `engine` as a supported argument.  However if I create a resource with an engine argument, I get the error message:

```
* aws_rds_cluster.snappy_cluster: "engine": this field cannot be set
```

If `engine` is not a supported input argument, then the documentation for `aws_rds_cluster` shouldn't list it as an argument.  

The reason I want to specify the engine argument is because AWS recently added support for PostgresQL Aurora clusters.  I wanted to try specifying `engine = "aurora-postgresql"` as an argument to `aws_rds_cluster` to create a PostgreSQL cluster.

### Terraform Version

```
$ terraform -v
Terraform v0.10.7
```

### Terraform Configuration Files
```hcl
resource "aws_rds_cluster" "snappy_cluster" {
  cluster_identifier                  = "test-cluster"
  database_name                       = "testdb"
  master_username                     = "administrator"
  master_password                     = "adminpwd"
  backup_retention_period             = 1
  iam_database_authentication_enabled = true
  engine                              = "aurora-postgresql"
}

resource "aws_rds_cluster_instance" "snappy_cluster_instance" {
  cluster_identifier   = "${aws_rds_cluster.snappy_cluster.cluster_identifier}"
  identifier           = "test"
  instance_class       = "db.r4.large"
  publicly_accessible  = false
}
```

### Expected Behavior
Create an RDS cluster using the PostgresQL engine

### Actual Behavior
I get an error from terraform saying

```
* aws_rds_cluster.snappy_cluster: "engine": this field cannot be set
```

### Steps to Reproduce
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform plan`

**Comment by bflad** [comment]
Dupe of #1443

**Comment by Ninir** [comment]
Hi @ajenkins-cargometrics,

Thank you for opening this issue. Even though you are using the latest Terraform version, providers now use their own repository, issues but most importantly their own versioning.

I assume you are using an old version of the AWS provider. To ensure & confirm that, can you go to the stack you are actually trying to deploy, remove the `.terraform/plugins` directory and then run `terraform init` again? it will download the latest AWS provider version and you should be able to use the `engine` parameter 🚀 

Tell us how it goes! :)

**Comment by kchandra** [comment]
@Ninir, I was having similar issues as @ajenkins-cargometrics. Followed your steps and it worked. Thanks

**Comment by bflad** [comment]
This was previously released in [terraform-provider-aws version 1.0.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#100-september-27-2017) and has been available in all releases since. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
